tests/unit/CoreListTest.php

```diff
@@ -0,0 +1,553 @@
+								'right'  => 'var:preset|spacing|30',
+							],
+						],
+					]
+				),
+				'textColor'       => 'accent-3',
+				'type'            => 'upper-alpha',
+				'values'          => '<li>Pizza</li><li>Pasta</li>',
+			],
+			$block['attributes'],
+		);
+	}
+
+	/**
+	 * Test case for retrieving core list block fields and attributes.
+	 *
+	 * Covers : 'className', 'content', 'fontSize', 'fontFamily' and 'lock' attributes.
+	 */
+	public function test_retrieve_core_list_item_fields_and_attribute(): void {
+		$block_content = '
+			<!-- wp:list -->
+				<ul class="wp-block-list">
+					<!-- wp:list-item {"lock":{"move":true,"remove":true},"className":"test-css-class-item-1","fontSize":"large","fontFamily":"heading"} -->
+						<li class="test-css-class-item-1 has-heading-font-family has-large-font-size">List item 1</li>
+					<!-- /wp:list-item -->
+				</ul>
+			<!-- /wp:list -->
+		';
+
+		// Set post content.
+		wp_update_post(
+			[
+				'ID'           => $this->post_id,
+				'post_content' => $block_content,
+			]
+		);
+
+		$query     = $this->query();
+		$variables = [
+			'id' => $this->post_id,
+		];
+
+		$actual = graphql( compact( 'query', 'variables' ) );
+
+		$this->assertArrayNotHasKey( 'errors', $actual, 'There should not be any errors' );
+		$this->assertArrayHasKey( 'data', $actual, 'The data key should be present' );
+		$this->assertArrayHasKey( 'post', $actual['data'], 'The post key should be present' );
+
+		$this->assertEquals( $this->post_id, $actual['data']['post']['databaseId'], 'The post ID should match' );
+		$this->assertEquals( 1, count( $actual['data']['post']['editorBlocks'] ) );
+
+		$block = $actual['data']['post']['editorBlocks'][0]['innerBlocks'][0];
+
+		$this->assertNotEmpty( $block['apiVersion'], 'The apiVersion should be present' );
+		$this->assertEquals( 'text', $block['blockEditorCategoryName'], 'The blockEditorCategoryName should be text' );
+		$this->assertNotEmpty( $block['clientId'], 'The clientId should be present' );
+		$this->assertNotEmpty( $block['cssClassNames'], 'The cssClassNames should be present' );
+		$this->assertEquals( 'core/list-item', $block['name'], 'The block name should be core/list' );
+
+		// @todo : The 'parentClientId' attribute is NULL in the response.
+		// $this->assertNotEmpty( $block['parentClientId'], 'There should be some parentClientId for the block' );
+
+		$this->assertNotEmpty( $block['renderedHtml'], 'The renderedHtml should be present' );
+		$this->assertEquals(
+			[
+				'className'   => 'test-css-class-item-1',
+				'content'     => 'List item 1',
+				'fontFamily'  => 'heading',
+				'fontSize'    => 'large',
+				'lock'        => wp_json_encode(
+					[
+						'move'   => true,
+						'remove' => true,
+					]
+				),
+				'placeholder' => null,
+				'style'       => null,
+			],
+			$block['attributes'],
+		);
+	}
+
+	/**
+	 * Test case for retrieving core list item block fields and attributes.
+	 *
+	 * Covers : 'placeholder' and 'style' attribute.
```

Comment:
We're not testing placeholder if it comes back null. Remove from here and the below comment. and then just rename the class to indicate its just testing the list item style attribute.

Reply:
Done ✅